File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/dto/rest/Page.java
package com.raxim.myscoutee.profile.data.dto.rest;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class Page<T> {
    @JsonProperty(value = "values")
    private List<T> values;

    @JsonIgnore
    private List<Object> offset;

    @JsonProperty(value = "scroll")
    private Integer scroll;

    @JsonProperty(value = "step")
    private Object step;

    @JsonProperty(value = "role")
    private Object role;

    public List<T> getValues() {
        return values;
    }

    public void setValues(List<T> values) {
        this.values = values;
    }

    public List<Object> getOffset() {
        return offset;
    }

    public void setOffset(List<Object> offset) {
        this.offset = offset;
    }

    public Integer getScroll() {
        return scroll;
    }

    public void setScroll(Integer scroll) {
        this.scroll = scroll;
    }

    public Object getStep() {
        return step;
    }

    public void setStep(Object step) {
        this.step = step;
    }

    public Object getRole() {
        return role;
    }

    public void setRole(Object role) {
        this.role = role;
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/dto/rest/Page.kt
package com.raxim.myscoutee.profile.data.dto.rest

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

data class Page<T>(
    @field:JsonProperty(value = "values")
    val values: List<T>? = emptyList(),

    @field:JsonProperty(value = "offset")
    val offset: List<Any>? = emptyList(),

    //scroll down only = 1
    @field:JsonProperty(value = "scroll")
    val scroll: Int? = 1,

    @field:JsonProperty(value = "step")
    val step: Any? = null,

    @field:JsonProperty(value = "role")
    val role: Any? = null

)